app/controllers/articles_controller.rb

```diff
@@ -18,7 +20,28 @@ def show
     end
 
     def index
-        @article = Article.all
+    
+
+        @last_page = (Article.all.count / ARTICLES_PER_PAGE).floor
+        
+        #fetch finds the value of the page query parameter, if there is none supplied use 0.
+        page_from_params = params.fetch(:page, 0).to_i
+        
+        #error handling to handle negative page/offset or when offset is greater than the last page.
+        if (page_from_params > -1) and (page_from_params < @last_page + 1)
+        
+            @page = page_from_params
+        elsif page_from_params < 0

```

Comment:
You added a default parameter to the fetch method. That means that if :page is falsey, :page will equal 0. So page_from_params could never be less than zero. https://apidock.com/ruby/Hash/fetch